dbms/src/Functions/divide.cpp

```diff
@@ -62,7 +62,10 @@ struct TiDBDivideFloatingImpl<A, B, false>
     template <typename Result = ResultType>
     static Result apply(A a, B b)
     {
-        return static_cast<Result>(a) / b;
+        if constexpr (std::is_integral_v<Result>)
+            return (static_cast<Result>(a) + b / 2) / b;
```

Comment:
Can you give some comments about why it was handled in this way?

Reply:
added